server/neptune/workflows/internal/deploy/revision/queue/deployer.go

```diff
@@ -84,15 +92,59 @@ func (p *Deployer) Deploy(ctx workflow.Context, requestedDeployment terraformWor
 		return nil, err
 	}
 
-	latestDeployment = requestedDeployment.BuildPersistableInfo()
-	if persistErr := p.persistLatestDeployment(ctx, latestDeployment); persistErr != nil {
-		logger.Error(ctx, "unable to persist deployment, proceeding with in-memory value", key.ErrKey, persistErr)
+	// log error and continue deploys if any of the post deploy task fails
+	if err := p.runPostDeployTasks(ctx, requestedDeployment, scope); err != nil {
+		logger.Error(ctx, "error running post deploy task", key.ErrKey, err)
 	}
 
 	// Count this as deployment as latest if it's not a PlanRejectionError which means it is a TerraformClientError
 	// We do this as a safety measure to avoid deploying out of order revision after a failed deploy since it could still
 	// mutate the state file
-	return latestDeployment, err
+	return requestedDeployment.BuildPersistableInfo(), err
+}
+
+func (p *Deployer) runPostDeployTasks(ctx workflow.Context, deployment terraform.DeploymentInfo, scope metrics.Scope) error {
+	if err := p.persistLatestDeployment(ctx, deployment.BuildPersistableInfo()); err != nil {
+		return errors.Wrap(err, "persisting deployment")
+	}
+
+	if err := p.startPRRevisionWorkflow(ctx, deployment, scope); err != nil {
+		return errors.Wrap(err, "starting PR Revision workflow")
+	}
+
+	return nil
+}
+
+func (p *Deployer) startPRRevisionWorkflow(ctx workflow.Context, deployment terraform.DeploymentInfo, scope metrics.Scope) error {
+	version := workflow.GetVersion(ctx, version.SetPRRevision, workflow.DefaultVersion, 1)
+	if version == workflow.DefaultVersion {
+		return nil
+	}
+
+	ctx = workflow.WithChildOptions(ctx, workflow.ChildWorkflowOptions{
+		TaskQueue: prrevision.TaskQueue,
+		RetryPolicy: &temporal.RetryPolicy{
+
+			// starting pr revision wf shouldn't take more than 3 tries unless the worker is down, which we'll get alarmed on to fix
+			MaximumAttempts: PRRevisionWorkflowRetryCount,
+		},
+
+		// configuring this ensures the child workflow will continue execution when the parent workflow terminates
+		ParentClosePolicy: enums.PARENT_CLOSE_POLICY_ABANDON,
+	})
+
+	future := workflow.ExecuteChildWorkflow(ctx, p.PRRevisionWorkflow, prrevision.Request{
+		Repo:     deployment.Repo,
+		Root:     deployment.Root,
+		Revision: deployment.Revision,
+	})
+
+	var childWE workflow.Execution
+	if err := future.GetChildWorkflowExecution().Get(ctx, &childWE); err == nil {
```

Comment:
Also this condition says if err == nil instead of err != nil.  Might be worth logging the runid after we've started it btw.

Reply:
Fixed the error condition. 

> Might be worth logging the runid after we've started it btw.

The event history has the run id of the child workflow ([example](https://temporal-staging-internal.lyft.net/namespaces/atlantis/workflows/lyft%252Forchestration-sandbox-neptune%7C%7Csimple3/aa18b439-5792-4efa-9dea-ce09ae7e63ff/history/compact?sort=descending)) btw. I don't see value in logging this since we'll likely use the event history to navigate to the PR Revision workflow when debugging. Lmk if you think otherwise.